cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/upgrade/TemplateSetUpgraderTest.java

```diff
@@ -37,121 +37,122 @@
+
+  @Before
+  public void prepare() throws IOException {
+
+    Path playground = this.tempFolder.newFolder(".cobigen").toPath();
+    FileUtils.copyDirectory(new File(testFileRootPath + "valid-2.1"), playground.toFile());
+    this.templateLocation = playground.resolve(ConfigurationConstants.TEMPLATES_FOLDER);
+  }
+
+  /**
+   * Test the correct folder creation
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTemplateSetUpgrade() throws Exception {
+
+    TemplateSetUpgrader templateSetUpgrader = new TemplateSetUpgrader();
+    templateSetUpgrader.upgradeTemplatesToTemplateSets(this.templateLocation);
+
+    Path templateSetsPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    Path templateSetsAdapted = templateSetsPath.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+    assertThat(templateSetsPath).exists();
+    assertThat(templateSetsAdapted).exists();
+
+  }
+
+  /**
+   * Tests if the Template files are copied correct into the new template set and into the backup folder
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTemplateSetUpgradeCopyOfTemplates() throws Exception {
+
+    Path oldTemplatesPath = this.templateLocation.resolve(ConfigurationConstants.COBIGEN_TEMPLATES)
+        .resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+    int OldTemplatesFileCount = oldTemplatesPath.toFile().list().length;
+    Set<String> OldPathFilesSet = new HashSet<>(Arrays.asList(oldTemplatesPath.toFile().list()));
+
+    TemplateSetUpgrader templateSetUpgrader = new TemplateSetUpgrader();
+    templateSetUpgrader.upgradeTemplatesToTemplateSets(this.templateLocation);
+
+    Path backupPath = this.templateLocation.getParent().resolve("backup")
+        .resolve(ConfigurationConstants.TEMPLATES_FOLDER).resolve(ConfigurationConstants.COBIGEN_TEMPLATES)
+        .resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+    Set<String> backupPathFilesSet = new HashSet<>(Arrays.asList(backupPath.toFile().list()));
+    Path newTemplatesPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER)
+        .resolve(ConfigurationConstants.ADAPTED_FOLDER);
+    Set<String> NewPathFilesSet = new HashSet<>(Arrays.asList(newTemplatesPath.toFile().list()));
+
+    assertEquals(OldTemplatesFileCount - 1, NewPathFilesSet.size());
+    assertEquals(OldTemplatesFileCount, backupPathFilesSet.size());
+
+    for (String s : OldPathFilesSet) {
+      if (!s.equals("context.xml")) {
+        assertTrue(NewPathFilesSet.contains(s));
+        NewPathFilesSet.remove(s);
+      }
+      assertTrue(backupPathFilesSet.contains(s));
+      backupPathFilesSet.remove(s);
+
+    }
+    assertEquals(NewPathFilesSet.size(), 0);
+    assertEquals(backupPathFilesSet.size(), 0);
+  }
+
+  /**
+   * Tests the correct location for the created context.xml and if the files and whether the file corresponds to the
+   * v3.0 schema
```

Comment:
Better: Tests the correct location for the created context.xml and whether the file corresponds to the v3.0 schema